### PR TITLE
Documentation of `tf.nn.depthwise_conv2d` Add limitation doc on strides

### LINKED ISSUE #62271: Documentation of `tf.nn.depthwise_conv2d` fails to mention limitation on strides
### Issue type

Documentation Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.14.0

### Custom code

No

### OS platform and distribution

macOS 13.6

### Mobile device

none

### Python version

3.11

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

Apparently, tensorflow's [`tf.nn.depthwise_conv2d`](https://www.tensorflow.org/api_docs/python/tf/nn/depthwise_conv2d) recently lost the ability to work with inequal strides (see #60391; apparently it was supported up to TF 2.11). This is mentioned in the documentation of [`tf.keras.layers.SeparableConv2D`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/SeparableConv2D) (albeit IMHO it should be more emphasised given that this is a limitation specific to depthwise convolutions).

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
layer1 = tf.nn.depthwise_conv2d(np.ones((2,3,4,5)),filter=np.ones((1,2,5)),strides=(1,1,2,1),padding="SAME")
```

### Relevant log output

```shell
InvalidArgumentError: {{function_node __wrapped__DepthwiseConv2dNative_device_/job:localhost/replica:0/task:0/device:CPU:0}} Current implementation only supports equal length strides in the row and column dimensions. [Op:DepthwiseConv2dNative] name:
```

### PR DESCRIPTION
**Pull request by rajveer43**
fixes #62271

### CODE REVIEW

File: tensorflow/python/ops/nn_impl.py [hunk 1]
```diff
@@ -817,9 +817,9 @@ def depthwise_conv2d_v2(input,
 
   Must have `strides[0] = strides[3] = 1`.  For the most common case of the
   same horizontal and vertical strides, `strides = [1, stride, stride, 1]`.
-  If any value in `dilations` is greater than 1, we perform atrous depthwise
-  convolution, in which case all values in the `strides` tensor must be equal
-  to 1.
+  Please note that if any value in `dilations` is greater than 1, we perform
```
Comment (cantonios) [inline tensorflow/python/ops/nn_impl.py:820]: Why this change?

File: tensorflow/python/ops/nn_impl.py [hunk 2]
```diff
@@ -837,17 +837,17 @@ def depthwise_conv2d_v2(input,
     array([[[[10., 14.],
              [14., 20.]],
             [[18., 26.],
-             [22., 32.]]]], dtype=float32)
+             [22., 32.]]]], dtype float32)
```
Comment (cantonios) [inline tensorflow/python/ops/nn_impl.py:840]: This is incorrect.

File: tensorflow/python/ops/nn_impl.py [hunk 3]
```diff
@@ -837,17 +837,17 @@ def depthwise_conv2d_v2(input,
...
 
   >>> tf.nn.depthwise_conv2d(x, kernel, strides=[1, 1, 1, 1],
   ...                        padding=[[0, 0], [1, 0], [1, 0], [0, 0]]).numpy()
     array([[[[ 0.,  0.],
              [ 3.,  4.],
-             [ 6.,  8.]],
+             [ 6., 8.]],
```
Comment (cantonios) [inline tensorflow/python/ops/nn_impl.py:846]: Please revert formatting.

File: tensorflow/python/ops/nn_impl.py [hunk 4]
```diff
@@ -856,7 +856,9 @@ def depthwise_conv2d_v2(input,
     filter: 4-D with shape
       `[filter_height, filter_width, in_channels, channel_multiplier]`.
     strides: 1-D of size 4.  The stride of the sliding window for each
-      dimension of `input`.
+      dimension of `input`. Must have `strides[0] = strides[3] = 1`, and for
```
Comment (cantonios) [inline tensorflow/python/ops/nn_impl.py:859]: This is already described above in the details section.  Do you really think it's necessary to have here too?

**Review by cantonios — CHANGES_REQUESTED** [review]

**Comment by rajveer43** [comment]
The issue mentioned that there is `strides` documentation is not proper

**Comment by cantonios** [comment]
> The issue mentioned that there is `strides` documentation is not proper

Ah, I see, the documentation is a bit unclear in the above section.  Sure, you can fix that.  You'll need to update the PR though.

File: tensorflow/python/ops/nn_impl.py [hunk 5]
```diff
@@ -856,9 +856,7 @@ def depthwise_conv2d_v2(input,
     filter: 4-D with shape
       `[filter_height, filter_width, in_channels, channel_multiplier]`.
     strides: 1-D of size 4.  The stride of the sliding window for each
-      dimension of `input`. Must have `strides[0] = strides[3] = 1`, and for
```
Comment (cantonios) [inline tensorflow/python/ops/nn_impl.py]: You'll need to re-add something here, but you need to explicitly mention that _only_ equal strides are supported (I'm guessing only if `strides[0]` and `strides[3]` are 1?).  Do some testing to confirm, and fix where it's mentioned above as well.

Comment (cantonios) [inline tensorflow/python/ops/nn_impl.py]: You'll need to re-add something here, but you need to explicitly mention that _only_ equal strides are supported (I'm guessing only if `strides[0]` and `strides[3]` are 1?).  Do some testing to confirm, and fix where it's mentioned above as well.

**Review by cantonios — CHANGES_REQUESTED** [review]

**Comment by rajveer43** [comment]
@cantonios, sure I will do some testing for it! and then add it

**Comment by gbaned** [comment]
Hi @rajveer43 Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @rajveer43 Any update on this PR? Please. Thank you!

**Pull request closed without merging**